simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/HotspotMetricCollectorsConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import io.prometheus.client.Summary;
+import io.prometheus.client.hotspot.ClassLoadingExports;
+import io.prometheus.client.hotspot.GarbageCollectorExports;
+import io.prometheus.client.hotspot.MemoryPoolsExports;
+import io.prometheus.client.hotspot.StandardExports;
+import io.prometheus.client.hotspot.ThreadExports;
+
+/**
+ * A {@link Configuration} class that imports the Prometheus
+ * JVM management metrics Collectors from the <code>simpleclient_hotspot</code> package.
+ *
+ * @author pidster
+ */
+@Configuration
+public class HotspotMetricCollectorsConfiguration {
```

Comment:
I'm currently integrating Prometheus in our Spring Boot app and am experiencing pain the static stuff causes. As @pidster pointed out, static state is bad. One of the reasons why you use DI frameworks such as Spring is to get rid of static state. 

You cannot use Spring Devtools with the current static implementation. Spring tries to reload the context but the static stuff stays around. As a result, the context does not come back up because already registered metrics survive the reload. I have to call `clear()` to get around this. Clearing the registry then causes the next problem. The default exports cannot be re-registered because they track initialization in a static field which is still `true` because it also survives the reload. As a result, I have this in my Spring configuration which  is ugly:

```java
@Autowired
void initializeDefaultsExports() {
    // The default registry, which is also used by the Spring Boot integration, 
    // is static. It survives a reload with Spring Devtools. As a result the 
    // Spring context does not come back up again, because the registry  
    // throws an exception because we try to register metrics which are 
    // already there. Thus, we just clear the registry to be on the safe side.
    CollectorRegistry.defaultRegistry.clear();

    // We cannot use this: DefaultExports.initialize()
    // It tracks initialization which is still true after a reload with Spring  
    // Devtools causing the exports not to be re-registered.
    new StandardExports().register();
    new MemoryPoolsExports().register();
    new GarbageCollectorExports().register();
    new ThreadExports().register();
    new ClassLoadingExports().register();
}
```

Thanks for your effort, @pidster. I'd very much like to see a solution that let's me opt out from static state.

@brian-brazil Could you please consider looking at this again? Maybe we can go for a simpler solution with less Spring magic, if that suits you better. All I ask for is to be able to provide a non-static CollectorRegistry. I'd be happy to help.